maintenance.md

```diff
@@ -0,0 +1,518 @@
+4. Update image tag in `README.md`.
+5. Update `BRANCH` and `TAG` files.
+
+## configmap-reload
+
+1. Check the [release page](https://github.com/jimmidyson/configmap-reload/releases).
+2. Check the upstream Dockerfile. If there are any updates, update our `Dockefile`.
+   - https://github.com/jimmidyson/configmap-reload/blob/vX.Y.Z/Dockerfile
+3. Update `CONFIGMAP_RELOAD_VERSION` in `Dockerfile`
+4. Update `src/CHANGELOG.md`.
+5. Update `BRANCH` and `TAG` files.
+
+## contour
+
+![Regular Update](./regular_update.svg)
+
+1. Check the [release page](https://github.com/projectcontour/contour/releases).
+2. Check the upstream Dockerfile. If there are any updates, update our `Dockefile`.
+   - https://github.com/projectcontour/contour/blob/vX.Y.Z/Dockerfile
+3. Update `CONTOUR_VERSION` in `Dockerfile`.
+4. Update image tag in `README.md`.
+5. Update `BRANCH` and `TAG` files.
+
+## coredns
+
+![Kubernetes Update](./kubernetes_update.svg)
+
+1. Check the [release page](https://github.com/coredns/coredns/releases).
+2. Check the upstream Dockerfile. If there are any updates, update our `Dockefile`.
+   - https://github.com/coredns/coredns/blob/vX.Y.Z/Dockerfile
+3. Update `COREDNS_VERSION` in `Dockerfile`.
+4. Update image tag in `README.md`.
+5. Update `BRANCH` and `TAG` files.
+
+## dex
+
+![Regular Update](./regular_update.svg)
+
+***NOTE:*** This image is used by ArgoCD. So browse the following manifest and check the required version.
+- https://github.com/argoproj/argo-cd/blob/vX.Y.Z/manifests/base/dex/argocd-dex-server-deployment.yaml
+
+1. Check the [release page](https://github.com/dexidp/dex/releases).
+2. Check the upstream Dockerfile. If there are any updates, update our `Dockefile`.
+   - https://github.com/dexidp/dex/blob/vX.Y.Z/Dockerfile
+3. Update `DEX_VERSION` in `Dockerfile`.
+4. Update image tag in `README.md`.
+5. Update `BRANCH` and `TAG` files.
+
+## dnsmasq
+
+![Regular Update](./regular_update.svg)
+
+1. Check the [dnsmasq-base package](https://packages.ubuntu.com/bionic/dnsmasq-base) in the Ubuntu packages.
+2. Check the [CHANGELOG](http://www.thekelleys.org.uk/dnsmasq/CHANGELOG) in the official website, if the `dnsmasq-base` package is updated.
+4. Update `DNSMASQ_VERSION` in `Dockerfile`.
+5. Update image tag in `README.md`.
+6. Update `BRANCH` and `TAG` files.
+
+## envoy
+
+![Regular Update](./regular_update.svg)
+
+***NOTE:*** Envoy is managed by Contour so update to the supported version. See the below.
+- [Envoy Support Matrix](https://projectcontour.io/resources/envoy/)
+
+1. Check the [release page](https://github.com/envoyproxy/envoy/releases).
+2. Update the `version` argument on the `build-envoy` job in the CircleCI `main` workflow.
+3. Update `BAZEL_VERSION` in `build-envoy` job. The required version is written in the following file.
+   - https://github.com/envoyproxy/envoy/blob/vX.Y.Z/.bazelversion
+4. Update image tag in `README.md`.
+5. Update `BRANCH` and `TAG` files.
+
+## etcd
+
```

Comment:
I understand that both the Kubernetes badge and the regular update badge are not needed here, but it's better to write etcd is updated in a separated task.